src/libmcrx.c

```diff
@@ -1055,3 +1075,429 @@ mcrx_is_system_error(enum mcrx_error_code err) {
   return err >= MCRX_ERR_SYSCALL_BIND;
 }
 
+/**
+ * mcrx_mnatmap_new:
+ * @mnatmapp: pointer to be filled with the new mnat context
+ *
+ * Create a new mnat context.
+ *
+ * Returns: error code
+ **/
+MCRX_EXPORT enum mcrx_error_code mcrx_mnatmap_new(
+    struct mcrx_mnatmap **mnatmapp) {
+  if (mnatmapp == NULL) {
+    return MCRX_ERR_NULLARG;
+  }
+
+  struct mcrx_mnatmap *mnat_ctx;
+  mnat_ctx = calloc(1, sizeof(struct mcrx_mnatmap));
+  if (!mnat_ctx) {
+    return MCRX_ERR_NOMEM;
+  }
+
+  mnat_ctx->refcount = 1;
+  LIST_INIT(&mnat_ctx->mnats_head);
+
+  *mnatmapp = mnat_ctx;
+  return MCRX_ERR_OK;
+}
+
+/**
+ * mcrx_mnatmap_apply:
+ * Apply the mnat context to mcrx context
+ *
+ **/
+MCRX_EXPORT enum mcrx_error_code mcrx_mnatmap_apply(struct mcrx_ctx *ctx,
+    struct mcrx_mnatmap *mnatmap) {
```

Comment:
as in the review for the latest meeting, this function should refactor in a couple of ways:
- the cases should combine to be the same, regardless of whether the initial ctx->mnat_map was NULL or non-NULL
- calling mcrx_mnatmap_apply with a NULL input mnatmap should result in removing the mnatmap from the ctx (including removing the entries from all the subs)